### PR TITLE
router: Fix outlier detection false positives

### PR DESCRIPTION
**Pull request by Hexta**
Commit Message: Fix outlier detection false positives caused by max requests circuit breaker.
Risk Level: low
Testing: integration tests
Release Notes: Done.
Fixes: #25487

### CODE REVIEW

File: test/integration/circuit_breakers_integration_test.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class CircuitBreakersIntegrationTest : public HttpProtocolIntegrationTest {
+public:
+  void initialize() override { HttpProtocolIntegrationTest::initialize(); }
+};
+
+INSTANTIATE_TEST_SUITE_P(Protocols, CircuitBreakersIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams()),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+TEST_P(CircuitBreakersIntegrationTest, CircuitBreakersWithOutlierDetection) {
```
Comment (mattklein123) [inline test/integration/circuit_breakers_integration_test.cc:17]: Can you add some comment on what this test is testing as it's a little complicated?

File: source/common/router/router.cc [hunk 2]
```diff
@@ -1286,18 +1286,21 @@ void Filter::onUpstreamReset(Http::StreamResetReason reset_reason,
...
 
-  // TODO: The reset may also come from upstream over the wire. In this case it should be
-  // treated as external origin error and distinguished from local origin error.
-  // This matters only when running OutlierDetection with split_external_local_origin_errors
-  // config param set to true.
-  updateOutlierDetection(Upstream::Outlier::Result::LocalOriginConnectFailed, upstream_request,
-                         absl::nullopt);
+  const bool dropped = reset_reason == Http::StreamResetReason::Overflow;
+
+  if (!dropped) {
```
Comment (mattklein123) [inline source/common/router/router.cc:1291]: Can you add a small comment as to why we are not doing outlier detection for circuit breakers? I think this makes sense but it's worth it to add some comment to the code so the next person coming along has the history.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Generally LGTM, thanks. Some small comments. Also, I think this is worth a release note. Thank you.

/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Comment by Hexta** [comment]
@mattklein123 Thx for the review. 
Unfortunately, some integration test was failed.
As I see this test is failing periodically for unclear reasons.

```
test/integration/redirect_integration_test.cc:704
Expected equality of these values:
1
test_server_->counter("cluster.cluster_0.upstream_internal_redirect_succeeded_total") ->value()
Which is: 0
```

**Pull request merged**